*** src/BulletDynamics/Dynamics/btDiscreteDynamicsWorld.cpp	2022-12-15 13:08:36.000000000 -0800
--- src/BulletDynamics/Dynamics/btDiscreteDynamicsWorld.cpp-disable-sleeping	2022-12-15 08:52:32.000000000 -0800
***************
*** 595,631 ****
  {
  	BT_PROFILE("updateActivationState");
  
! 	for (int i = 0; i < m_nonStaticRigidBodies.size(); i++)
! 	{
! 		btRigidBody* body = m_nonStaticRigidBodies[i];
! 		if (body)
! 		{
! 			body->updateDeactivation(timeStep);
! 
! 			if (body->wantsSleeping())
! 			{
! 				if (body->isStaticOrKinematicObject())
! 				{
! 					body->setActivationState(ISLAND_SLEEPING);
! 				}
! 				else
! 				{
! 					if (body->getActivationState() == ACTIVE_TAG)
! 						body->setActivationState(WANTS_DEACTIVATION);
! 					if (body->getActivationState() == ISLAND_SLEEPING)
! 					{
! 						body->setAngularVelocity(btVector3(0, 0, 0));
! 						body->setLinearVelocity(btVector3(0, 0, 0));
! 					}
! 				}
! 			}
! 			else
! 			{
! 				if (body->getActivationState() != DISABLE_DEACTIVATION)
! 					body->setActivationState(ACTIVE_TAG);
! 			}
! 		}
! 	}
  }
  
  void btDiscreteDynamicsWorld::addConstraint(btTypedConstraint* constraint, bool disableCollisionsBetweenLinkedBodies)
--- 595,634 ----
  {
  	BT_PROFILE("updateActivationState");
  
! 	// for (int i = 0; i < m_nonStaticRigidBodies.size(); i++)
! // 	{
! // 		btRigidBody* body = m_nonStaticRigidBodies[i];
! // 		if (body)
! // 		{
! // 			body->updateDeactivation(timeStep);
! // 
! // 			if (body->wantsSleeping())
! // 			{
! // 				if (body->isStaticOrKinematicObject())
! // 				{
! // 					body->setActivationState(ISLAND_SLEEPING);
! // 				}
! // 				else
! // 				{
! // 					if (body->getActivationState() == ACTIVE_TAG)
! // 						body->setActivationState(WANTS_DEACTIVATION);
! // 					if (body->getActivationState() == ISLAND_SLEEPING)
! // 					{
! // 						body->setAngularVelocity(btVector3(0, 0, 0));
! // 						body->setLinearVelocity(btVector3(0, 0, 0));
! // 						// when sleeping, force the motion state to be called/updated
! // 						if (body->getMotionState())
! // 							body->getMotionState()->setWorldTransform(body->getWorldTransform());
! // 					}
! // 				}
! // 			}
! // 			else
! // 			{
! // 				if (body->getActivationState() != DISABLE_DEACTIVATION)
! // 					body->setActivationState(ACTIVE_TAG);
! // 			}
! // 		}
! // 	}
  }
  
  void btDiscreteDynamicsWorld::addConstraint(btTypedConstraint* constraint, bool disableCollisionsBetweenLinkedBodies)
